core/domain/suggestion_registry.py

```diff
@@ -2001,3 +2001,511 @@ def __init__(
+        self.recent_performance = recent_performance
+        self.overall_accuracy = overall_accuracy
+        self.submitted_translations_count = submitted_translations_count
+        self.submitted_translation_word_count = submitted_translation_word_count
+        self.accepted_translations_count = accepted_translations_count
+        self.accepted_translations_without_reviewer_edits_count = (
+            accepted_translations_without_reviewer_edits_count
+        )
+        self.accepted_translation_word_count = accepted_translation_word_count
+        self.rejected_translations_count = rejected_translations_count
+        self.rejected_translation_word_count = rejected_translation_word_count
+        self.first_contribution_date = first_contribution_date
+        self.last_contribution_date = last_contribution_date
+
+    def to_dict(self) -> TranslationSubmitterTotalContributionStatsDict:
+        """Returns a dict representation of a
+        TranslationSubmitterTotalContributionStats domain object.
+
+        Returns:
+            dict. A dict representation of a
+            TranslationSubmitterTotalContributionStats domain object.
+        """
+        return {
+            'language_code': self.language_code,
+            'contributor_id': self.contributor_id,
+            'topic_ids_with_translation_submissions': (
+                self.topic_ids_with_translation_submissions),
+            'recent_review_outcomes': self.recent_review_outcomes,
+            'recent_performance': self.recent_performance,
+            'overall_accuracy': self.overall_accuracy,
+            'submitted_translations_count': (
+                self.submitted_translations_count),
+            'submitted_translation_word_count': (
+                self.submitted_translation_word_count),
+            'accepted_translations_count': (
+                self.accepted_translations_count),
+            'accepted_translations_without_reviewer_edits_count': (
+                self
+                .accepted_translations_without_reviewer_edits_count),
+            'accepted_translation_word_count': (
+                self.accepted_translation_word_count),
+            'rejected_translations_count': (
+                self.rejected_translations_count),
+            'rejected_translation_word_count': (
+                self.rejected_translation_word_count),
+            'first_contribution_date': self.first_contribution_date,
+            'last_contribution_date': self.last_contribution_date
+        }
+
+    def to_frontend_dict(
+            self) -> TranslationSubmitterTotalContributionStatsFrontendDict:
+        """Returns a dict representation of a
+        TranslationSubmitterTotalContributionStats domain object for frontend.
+
+        Returns:
+            dict. A dict representation of a
+            TranslationSubmitterTotalContributionStats domain object for
+            frontend.
+        """
+        return {
```

Comment:
If we do need both the `to_dict` and `to_frontend_dict` methods, then might want to call `to_dict` here and update the values that need modification since most of the fields remain the same.

Reply:
We are changing values, as well as name and type of some keys in the dict. For example we are changing `last_contribution_date`(date) to `last_contributed_in_days`(int)
Also we won't require the `topic_ids_with_translation_submissions` in frontend while displaying stats.

Reply:
We are changing the value and type of `last_contribution_date`(date) to `last_contributed_in_days`(int). Since we cannot delete a key from a TypedDict, we cannot use to_dict here and modify the values.

Reply:
Ah, I see. It shouldn't be a problem anymore anyway since we should just remove the `to_dict` methods.